src/test/java/uk/gov/justice/log/utils/PropertyReaderTest.java

```diff
@@ -0,0 +1,147 @@
+        mockSetupForSearchCriteria(Arrays.asList("key1"), Arrays.asList("[2][0][2]", "[4][0][0]"), 0, FROM, TO);
+        final PropertyReader propertyReader = new PropertyReader(setUpSearchParameters(CONFIG_PATH, SEARCH_PATH, "/Users/user.json", RESPONSE_PATH));
+
+        assertThat(propertyReader.searchCriteria().getRegexes().size(), is(2));
+        assertThat("User list file is not present at given location: " + "/Users/user.json", is(propertyReader.errors().get(0).trim()));
+        assertThat(propertyReader.errors().size(), is(1));
+    }
+
+    @Test
+    public void shouldFailWhenConfigFileIsNotFound() throws IOException, ParseException {
+        mockSetupForConfig(HOST_NAME, HOST_SCHEME, 8080, 8000, PROXY_HOST);
+        mockSetupForSearchCriteria(Arrays.asList("key1"), Arrays.asList("[2][0][2]", "[4][0][0]"), 0, FROM, TO);
+        mockSetupForUserListFile();
+
+
+        final PropertyReader propertyReader = new PropertyReader(setUpSearchParameters("/xyz", SEARCH_PATH, USER_LIST_PATH, RESPONSE_PATH));
+
+        assertThat(propertyReader.searchCriteria().getRegexes().size(), is(2));
+        assertThat(propertyReader.errors().get(0).trim(), is("config.yaml is not present at given location: " + "/xyz"));
+        assertThat(propertyReader.errors().size(), is(1));
+    }
+
+    @Test
+    public void shouldPassWhenUserListFileIsFound() throws IOException, ParseException {
+
+        mockSetupForConfig(HOST_NAME, HOST_SCHEME, 8080, 8000, PROXY_HOST);
+        mockSetupForSearchCriteria(Arrays.asList("key1"), Arrays.asList("[2][0][2]", "[4][0][0]"), 0, FROM, TO);
+        final Map<String, String> userData = new HashMap<>();
+        userData.put("u1", "p1");
+        userData.put("u2", "p2");
+        mockSetupForUserListFile();
+
+        final PropertyReader propertyReader = new PropertyReader(setUpSearchParameters(CONFIG_PATH, SEARCH_PATH, USER_LIST_PATH, RESPONSE_PATH));
+        assertThat(propertyReader.errors().size(), is(0));
+    }
+
+    @Test
+    public void shouldPassWhenBothFilesFound() throws IOException, ParseException {
+        final PropertyReader propertyReader = new PropertyReader(setUpSearchParameters(CONFIG_PATH, SEARCH_PATH, null, RESPONSE_PATH));
+        assertThat(propertyReader.errors().size(), is(0));
+    }
+
+    protected void mockSetupForConfig(final String hostName,
+                                      final String hostScheme,
+                                      final Integer hostPort,
+                                      final int proxyPort,
+                                      final String proxyHost) throws IOException {
+
+        final Map<String, Object> data = new HashMap<>();
+        data.put("hostName", hostName);
+        data.put("hostScheme", hostScheme);
+        data.put("hostPort", hostPort);
+        data.put("proxyPort", proxyPort);
+        data.put("proxyHost", proxyHost);
+
+        new Yaml().dump(data, new FileWriter(CONFIG_PATH));
```

Comment:
Yes agreed now all moved to a system temp directories and deleted at the end of the test